mailpoet/mailpoet
# Add error messages for "blocked" keys to api key verification form [MAILPOET-5103]

## Description

This PR adds messages for cases when an API key loses access to premium or MSS, or both.
This may happen because of exceeding subscriber limits or email volume limits.

![Screenshot 2023-04-13 at 15 24 40](https://user-images.githubusercontent.com/1082140/231775475-b2be54ec-6506-460b-a25f-627b91fbb843.png)
![Screenshot 2023-04-13 at 15 15 38](https://user-images.githubusercontent.com/1082140/231775490-702a3521-50a3-4902-945b-4a4e46158b03.png)


## Code review notes

_N/A_

## QA notes

- Please verify the key activation form works as expected. It should still work as described in https://mailpoet.atlassian.net/browse/MAILPOET-4464
- When working on the ticket I noticed that, in some cases the success/error messages were different immediately after submitting the form and after reloading the page. I fixed that in https://github.com/mailpoet/mailpoet/commit/43e626669abec93b69951282db7ff00b2d730d98 so when verifying the form, please also reload the settings page.
- The Key validation form is also used in the last step of the onboarding wizard.

We need to release the changes in the plugin before we release the changes in the shop, so I can't provide you with a testing API key. But you can emulate the blocked key response by changing a code in Bridge API. 

1. Open mailpoet/lib/Services/Bridge/API.php file
2. Alter the `checkKey` method: 
     a) `return ['code' => 403, 'data' => null, 'error_message' => 'Email volume limit reached'];` to emulate key with limited access due email volume limits
     b) `return ['code' => 403, 'data' => null, 'error_message' => 'Subscribers limit reached'];` to emulate key with exceeded subscriber limits


## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5103]

## After-merge notes

_N/A_


[MAILPOET-5103]: https://mailpoet.atlassian.net/browse/MAILPOET-5103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
